ntoskrnl/kd64/kdprint.c

```diff
@@ -343,31 +343,16 @@ KdpPrint(
     STRING OutputString;
     CHAR CapturedString[512];
 
-    /* Assume failure */
-    *Handled = FALSE;
-
-#if (NTDDI_VERSION >= NTDDI_VISTA)
-    if ((ComponentId >= KdComponentTableSize) && (ComponentId < MAXULONG))
-    {
-        /* Use the default component ID */
-        Mask = &Kd_DEFAULT_Mask;
-        // Level = DPFLTR_INFO_LEVEL; // Override the Level.
-    }
-#endif
-    /* Convert Level to bit field if required */
-    if (Level < 32) Level = 1 << Level;
-    Level &= ~DPFLTR_MASK;
-
-    /* Validate the mask */
-    if (!(Kd_WIN2000_Mask & Level) ||
```

Comment:
I will recheck whether this OR operator is really what we want, or should be AND instead... (and I'll do the same for the NtQueryDebugFilterState() function).

Reply:
This seems illogical to have the same code spread across two functions. Even if Windows doesn't do this call (by the way in release build this call may be inlined anyway)